File: Javascript/Algorithms/1128.js
let numEquivDominoPairs = dominoes => {
  let counter = Array(73).fill(0);
  // Defines a (not so perfect) mapping between dominoes to indices.
  const index = (i, j) => (9 * (i - 1)) - i + j;
  dominoes.forEach(([x, y]) => {
    let [i, j] = x > y ? [y, x] : [x, y];
    counter[index(i, j)] += 1;
  });
  let sum = 0;
  counter.filter(x => x).forEach(count => {
    sum += (count * (count - 1));
  })
  return Math.floor(sum / 2);
};

File: C/Algorithms/1128.c
#define INDEX(i, j) (9 * (i - 1) - i + j)
#define COUNT_SIZE 73

int numEquivDominoPairs(int **dominoes, int dominoesSize,
                        int *dominoesColSize) {
  uint counter[COUNT_SIZE] = {0};
  // Count dominoes.
  for (int i = 0; i < dominoesSize; i++) {
    int a = dominoes[i][0], b = dominoes[i][1];
    if (a > b) {
      counter[INDEX(b, a)]++;
    } else {
      counter[INDEX(a, b)]++;
    }
  }

  // sum pairs
  uint sum = 0;
  for (int i = 0; i < COUNT_SIZE; i++) {
    int x = counter[i];
    if (x)
      sum += (x * (x - 1));
  }
  return sum / 2;
}